Code/Framework/AzCore/AzCore/Script/ScriptProperty.cpp

```diff
@@ -15,6 +15,49 @@
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptProperty, "ScriptProperty", "{D227D737-F1ED-4FB3-A1FB-38E4985D2E7A}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptProperty);
+
+    AZ_TYPE_INFO_WITH_NAME_IMPL(FunctionalScriptProperty, "FunctionalScriptProperty", "{57D7418D-6B14-4A02-B50E-2E409D23CFC6}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(FunctionalScriptProperty, ScriptProperty);
+
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyNil, "ScriptPropertyNil", "{ACAD23F6-5E75-460E-BD77-1B477750264F}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyNil, ScriptProperty);
+
+    // ScriptPropertyBoolean is serialized to a prefab file using the type name of AzFramwork::ScriptPropertyBoolean
+    // So make sure the old name is being used
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyBoolean, "AzFramework::ScriptPropertyBoolean", "{EA7335F8-5B9F-4744-B805-FEF9240451BD}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyBoolean, ScriptProperty);
+
+    // ScriptPropertyNumber is serialized to a prefab file using the type name of AzFramwork::ScriptPropertyNumber
+    // So make sure the old name is being used
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyNumber, "AzFramework::ScriptPropertyNumber", "{5BCDFDEB-A75D-4E83-BB74-C45299CB9826}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyNumber, ScriptProperty);
+
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyString, "ScriptPropertyString", "{A0229C6D-B010-47E7-8985-EE220FC7BFAF}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyString, ScriptProperty);
+
+    // ScriptPropertyGenericClass is serialized to a prefab file using the type name of AzFramwork::ScriptPropertyGenericClass
+    // So make sure the old name is being used
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyGenericClass, "AzFramework::ScriptPropertyGenericClass", "{80618224-814C-44D4-A7B8-14B5A36F96ED}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyGenericClass, FunctionalScriptProperty);
+
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyNumberArray, "ScriptPropertyNumberArray", "{76609A01-46CA-442E-8BA6-251D529886AF}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyNumberArray, ScriptProperty);
+
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyBooleanArray, "ScriptPropertyBooleanArray", "{3A83958C-26C7-4A59-B6D7-A7805B0EC756}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyBooleanArray, ScriptProperty);
+
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyStringArray, "ScriptPropertyStringArray", "{899993A5-D717-41BB-B89B-04A27952CA6D}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyStringArray, ScriptProperty);
+
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyGenericClassArray, "ScriptPropertyGenericClassArray", "{28E986DD-CF7C-404D-9BEE-EEE067180CD1}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyGenericClassArray, ScriptProperty);
+
+    AZ_TYPE_INFO_WITH_NAME_IMPL(ScriptPropertyAsset, "ScriptPropertyAsset", "{4D4B7176-A6E1-4BB9-A7B0-5977EC724CCB}");
+    AZ_RTTI_NO_TYPE_INFO_IMPL(ScriptPropertyAsset, ScriptProperty);
```

Comment:
I believe these should have AZ:: in front of their names.
https://github.com/o3de/o3de/blob/152bc0a1851d881fe735adf54ec93e1ad7875b11/Code/Framework/AzCore/AzCore/Script/ScriptProperty.h#L420